runtime/doc/treesitter.txt

```diff
@@ -441,6 +499,40 @@ node_contains({node}, {range})                               *node_contains()*
     Return: ~
         (boolean) True if the node contains the range
 
+node_to_lsp_range({node})                                *node_to_lsp_range()*
+    Gets the node range in the lsp range format
+
+    Parameters: ~
+        {node}  (table) The node to convert
+
+    Return: ~
+        { start =`start_position`, end = `end_position` }
+
+                                                                *swap_nodes()*
+swap_nodes({bufnr}, {node_or_range1}, {node_or_range2}, {opts})
+    Swaps the two given nodes
+
+    Parameters: ~
+        {bufnr}                  (number) The buffer number
+        {node_or_range1}         (table) The 1st node or its range
+        {node_or_range2}         (table) The 2nd node or its range
+        {opts}                   (table) Options table
+        {opts.cursor_to_second}  (boolean) (default false) set cursor on the
+                                 second node after the swap
+
+update_selection({bufnr}, {node}, {opts})                 *update_selection()*
+    Sets visual selection for node
```

Comment:
this needs a bit more explanation. Is it setting visual selection based on the parsed scope represented by a TS node? Then we don't need this. We need : 

1. a function that sets visual selection from a "region" (related: https://github.com/neovim/neovim/pull/13896 )
2. a function that gets a "region" from a TS node

Reply:
Yes, that is a function we're probably not going to upstream (in this form at least).

(2. is already available as `get_node_range`)